soh/src/overlays/actors/ovl_En_Ms/z_en_ms.c

```diff
@@ -142,11 +150,18 @@ void EnMs_Talk(EnMs* this, GlobalContext* globalCtx) {
 
 void EnMs_Sell(EnMs* this, GlobalContext* globalCtx) {
     if (Actor_HasParent(&this->actor, globalCtx)) {
-        Rupees_ChangeBy(-sPrices[BEANS_BOUGHT]);
+        Rupees_ChangeBy((gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_SHUFFLE_MAGIC_BEANS)) ? -60 : -sPrices[BEANS_BOUGHT]);
         this->actor.parent = NULL;
-        this->actionFunc = EnMs_TalkAfterPurchase;
+        this->actionFunc =
+            (gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_SHUFFLE_MAGIC_BEANS)) ? EnMs_Wait : EnMs_TalkAfterPurchase;
     } else {
-        func_8002F434(&this->actor, globalCtx, GI_BEAN, 90.0f, 10.0f);
+        if (gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_SHUFFLE_MAGIC_BEANS)) {
+            GiveItemEntryFromActor(&this->actor, globalCtx,
+                                   Randomizer_GetItemFromKnownCheck(RC_ZR_MAGIC_BEAN_SALESMAN, GI_BEAN), 90.0f, 10.0f);
+            BEANS_BOUGHT = 10;
```

Comment:
Is BEANS_BOUGHT here a macro? I’m not seeing it defined in this file

Reply:
I see now that it is.

Any particular reason to handle  the incrementing here rather than within `Randomizer_Item_Give`,  which would be more consistent with how the vanilla option is handled (in `Item_Give`)?

Reply:
Actually now that I think about it, we probably do want to handle it within `Randomizer_Item_Give`,  then it would handle getting it from Impa song check on Skip Child Zelda

Reply:
BEANS_BOUGHT is basically being used as a flag for having bought the item from the Magic Bean Salesman, we don't want to set BEANS_BOUGHT for receiving the bean pack.

Reply:
Aha that makes sense, and I realize now that you are handling it in `Randomizer_Item_Give` already.